### DISCUSSION TITLE
糊了个关联启动的 PowerShell 脚本

### DISCUSSION THREAD

**Discussion by RtYkk** [Simple-Tracker/qBittorrent-ClientBlocker#32] [General]
这样无论 blocker & 本体放在任何地方，都没有影响了。
```ps1
# Open qbittorrent.exe
Start-Process "C:\$1\qBittorrent\qbittorrent.exe"

# Introduce a 5-second delay
Start-Sleep -Seconds 5

# Change directory to the location of qBittorrent-ClientBlocker
Set-Location "C:\$2\qBittorrent-ClientBlocker"

# Run qBittorrent-ClientBlocker
.\qBittorrent-ClientBlocker
```

这样开机启动是不是也更方便？如果可以要不我提个小小的 Pr?

**Comment by IAALAI** [comment]
看不出来有什么显著优点？感觉意义有限

**Comment by Simple-Tracker** [comment]
并未看出此 Shell 的作用, $1 和 $2 仍然需要外部传入, 反而相比原有的一行命令增加了步骤. 还是说 $1 和 $2 是内置的系统变量? 另外相比原有的版本这一新版本并不会检测 qBCB 是否已启动, 因此可能会造成重复启动.
若以 Service 形式注册, 后台检测+不显示前台进程的话对开机启动会更加方便, 只不过有错误也较难告知.

**Reply by RtYkk** [reply]
$1 $2 是我故意留空的，就是想用的自己把目录填进去就行。填好之后就可以直接右键运行 .ps1 了。并不麻烦
重复启动这个是问题，我还没试过。不过一般我开了就不管了。
嘛主要原来快捷方式那个我有点没看明白…这个就放在 discussion 里面吧，当作一种 alternative.

**Reply by Simple-Tracker** [reply]
时隔已久, 我还是来解释下.
```C:\Windows\System32\cmd.exe /c "(tasklist | findstr qBittorrent-ClientBlocker || start C:\Users\Example\qBittorrent-ClientBlocker\qBittorrent-ClientBlocker.exe) && start     qbittorrent.exe"```

对于快捷方式, 它的工作路径默认是快捷方式内设定的 qBittorrent 的路径, 因此在启动 qBittorrent 时无需提供额外的路径.  
但运行 qBCB 则需要用户一次性修改路径为自己的实际路径, 保存快捷方式后就可一直使用.

tasklist 是获取进程列表, 随后将列表通过管道传递给 findstr, findstr 会寻找其中是否已经有启动我们的客户端.  
若匹配到, 则视作顺利执行, 后面的 OR ```||``` 即不会再执行.  
若未匹配到, 则视作执行失败, 后面的 OR 被执行, 即启动 qBCB 客户端.  
所以这一逻辑可以实现不重复启用客户端, 具体用例有: qBittorrent 未响应, 用户希望重启 qBittorrent 等.

最后, 无论如何都启动 qBittorrent, 这是原始快捷方式的行为, qBittorrent 会自行检测是否重复启动, 我们无需处理.

cmd /c 则主要是方便同时执行快捷方式本身可能无法做到的执行多条命令, 以期不需要额外文件的简易效果.

**Reply by RtYkk** [reply]
谢谢

**Comment by IAALAI** [comment]
但是如果在windows下需要开机启动的话，计划任务应该好一点吧。windows的计划任务里还可以设置启动顺序，当qbittorrent启动之后再调起qBCB
